### PR TITLE
Install nix in CI to build the spec docs

### PR DESCRIPTION
**Pull request by jasagredo**
# Description

I accidentally removed Nix, and [the build job](https://github.com/input-output-hk/ouroboros-network/actions/runs/4942509759/jobs/8843276426#step:8:17) didn't fail so this went unnoticed.

### CODE REVIEW

**Review by amesgen — APPROVED** [review]

**Review by coot — APPROVED** [review]
Thanks

**Comment by coot** [comment]
I'll merge it once we get the merge queue finished - currently it's broken.

**Pull request merged**
